flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/strategy/EagerSchedulingStrategyTest.java

```diff
@@ -0,0 +1,119 @@
+	 */
+	@Test
+	public void testStartScheduling() {
+		TestingSchedulingTopology testingSchedulingTopology = new TestingSchedulingTopology();
+		JobVertexID jobVertexID = new JobVertexID();
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 0));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 1));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 2));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 3));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 4));
+
+		TestingSchedulerOperations testingSchedulerOperations = new TestingSchedulerOperations();
+		EagerSchedulingStrategy schedulingStrategy = new EagerSchedulingStrategy(
+				testingSchedulerOperations,
+				testingSchedulingTopology);
+
+		schedulingStrategy.startScheduling();
+
+		assertEquals(1, testingSchedulerOperations.getScheduledVertices().size());
+
+		Collection<ExecutionVertexDeploymentOption> scheduledVertices = testingSchedulerOperations.getScheduledVertices().get(0);
+		assertEquals(5, scheduledVertices.size());
+		for (SchedulingExecutionVertex schedulingExecutionVertex : testingSchedulingTopology.getVertices()) {
+			assertThat(getExecutionVertexIdsFromDeployOptions(scheduledVertices), hasItem(schedulingExecutionVertex.getId()));
+		}
+	}
+
+	/**
+	 * Tests that eager scheduling strategy will restart all vertices needing restarted at same time.
+	 */
+	@Test
+	public void testRestartTasks() {
+		TestingSchedulingTopology testingSchedulingTopology = new TestingSchedulingTopology();
+		JobVertexID jobVertexID = new JobVertexID();
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 0));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 1));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 2));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 3));
+		testingSchedulingTopology.addSchedulingExecutionVertex(new TestingSchedulingExecutionVertex(jobVertexID, 4));
+
+		TestingSchedulerOperations testingSchedulerOperations = new TestingSchedulerOperations();
+		EagerSchedulingStrategy schedulingStrategy = new EagerSchedulingStrategy(
+				testingSchedulerOperations,
+				testingSchedulingTopology);
+
+		Set<ExecutionVertexID> toBeRestartedVertices1 = new HashSet<>();
+		toBeRestartedVertices1.add(new ExecutionVertexID(jobVertexID, 0));
+		toBeRestartedVertices1.add(new ExecutionVertexID(jobVertexID, 4));
+		schedulingStrategy.restartTasks(toBeRestartedVertices1);
+
+		assertEquals(1, testingSchedulerOperations.getScheduledVertices().size());
```

Comment:
I think it should be avoided to operate on the test objects after assertions have been done (see Clean Code by Robert C Martin, page 127). I would group all assertions into one block:

```
		Set<ExecutionVertexID> toBeRestartedVertices1 = new HashSet<>(Arrays.asList(
			new ExecutionVertexID(jobVertexID, 0),
			new ExecutionVertexID(jobVertexID, 4)));
		schedulingStrategy.restartTasks(toBeRestartedVertices1);

		Set<ExecutionVertexID> toBeRestartedVertices2 = new HashSet<>(Arrays.asList(
			new ExecutionVertexID(jobVertexID, 1),
			new ExecutionVertexID(jobVertexID, 2),
			new ExecutionVertexID(jobVertexID, 3)));
		schedulingStrategy.restartTasks(toBeRestartedVertices2);

		assertEquals(2, testingSchedulerOperations.getScheduledVertices().size());

		Collection<ExecutionVertexDeploymentOption> scheduledVertices1 = testingSchedulerOperations.getScheduledVertices().get(0);
		assertThat(getExecutionVertexIdsFromDeployOptions(scheduledVertices1), containsInAnyOrder(toBeRestartedVertices1.toArray()));

		Collection<ExecutionVertexDeploymentOption> scheduledVertices2 = testingSchedulerOperations.getScheduledVertices().get(1);
		assertThat(getExecutionVertexIdsFromDeployOptions(scheduledVertices2), containsInAnyOrder(toBeRestartedVertices2.toArray()));
```

Reply:
Done, good point